pkg/apis/agones/v1/gameserver.go

```diff
@@ -644,8 +644,11 @@ func (gs *GameServer) podObjectMeta(pod *corev1.Pod) {
 
 	if gs.Spec.Scheduling == apis.Packed {
 		// This means that the autoscaler cannot remove the Node that this Pod is on.
-		// (and evict the Pod in the process)
-		pod.ObjectMeta.Annotations["cluster-autoscaler.kubernetes.io/safe-to-evict"] = "false"
+		// (and evict the Pod in the process). Only set the value if it has not already
+		// been configured in the pod template (to not override user specified behavior).
+		if _, exists := pod.ObjectMeta.Annotations["cluster-autoscaler.kubernetes.io/safe-to-evict"]; !exists {
```

Comment:
Should "cluster-autoscaler.kubernetes.io/safe-to-evict" be a const? Might make it a bit easier to read.

As a possible other simplification (suggesting a possible const):

```go
if  _, exists := pod.ObjectMeta.Annotations[safeToEvicAnnotation]; !exists && gs.Spec.Scheduling == apis.Packed {
   pod.ObjectMeta.Annotations[safeToEvicAnnotation] = "false"
}
```

Make it one line? WDYT?

Reply:
I like that. I picked a name for the constant that was a bit longer but more closely matched the name used by the cluster autoscaler implementation (although they used "Key" instead of "Annotation" as the last part of their name which I found less descriptive).